src/utils/validateIdentifierInputs.ts

```diff
@@ -18,16 +18,27 @@ const isUniqueList = (list: string[]): boolean =>
 export const validateIdentifierInputs = (
   identifierInputs: IdentifierInput[]
 ): boolean => {
-  for (const { value, validationRegex, textInputType } of identifierInputs) {
-    if (textInputType === "SINGLE_CHOICE" && !value) {
-      throw new Error(ERROR_MESSAGE.MISSING_WAIVER_INPUT);
-    }
-    if (textInputType !== "PAYMENT_RECEIPT" && !value) {
-      throw new Error(ERROR_MESSAGE.MISSING_IDENTIFIER_INPUT);
+  for (const {
+    value,
+    validationRegex,
+    textInputType,
+    isOptional,
+  } of identifierInputs) {
+    if (isOptional && (!value || isNaN(Number(value)))) {
```

Comment:
Why was `isNaN(Number(value)))` added?

Also, shall we clean `value`, if it is not already done?

Reply:
i think value should already be cleaned before being passed to this function. under `checkoutCart`. i thought `isNaN` is a specific case to `NUMBER` to check for empty number. Should I just remove it?

Reply:
What happens when `value` is a string?

Reply:
shoot, u're right. changing